android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java

```diff
@@ -448,9 +445,26 @@ protected void onScrollChanged(int l, int t, int oldl, int oldt)
 				KrollDict data = new KrollDict();
 				getProxy().fireEvent(TiC.EVENT_DRAGSTART, data);
 			}
+
+			int contentWidth = 0;
+			int contentHeight = 0;
+
+			if (getChildCount() > 0) {
+				final View child = getChildAt(0);
+				TiDimension dimW = new TiDimension(child.getWidth(), TiDimension.TYPE_WIDTH);
+				TiDimension dimH = new TiDimension(child.getHeight(), TiDimension.TYPE_HEIGHT);
+				contentWidth = TiConvert.toInt(dimW.getAsDefault(child), 0);
+				contentHeight = TiConvert.toInt(dimH.getAsDefault(child), 0);
+			}
```

Comment:
Variables `contentWidth` and `contentHeight` should be of type `double` and not `int`. And we should not convert the `double` returned by `getAsDefault()` to an integer.

The reason is because the `contentSize` you are providing needs to use the units defined in the "tiapp.xml", which will typically be "dp" by default (not pixels). "dp" based values can be **fractional**. Especially on an "hdpi" Android device which has a `1.5` scaling factor. Test it out in an "hdpi" emulator and you'll see what I mean.

Reply:
done